src/lib.rs

```diff
@@ -192,6 +213,11 @@ macro_rules! impl_float_approx_eq {
             fn default_max_ulps() -> u32 { 4 }
 
             #[inline]
+            fn abs_diff_eq(&self, other: &$T, epsilon: $T) -> bool {
+                $T::abs(self - other) <= epsilon
```

Comment:
If this trait is intended to be possible to be implemented for unsigned integers, then you might run into underflow problems here. In `rulinalg` we handle this by a conditional on which element is bigger, see [here](https://github.com/AtheMathmo/rulinalg/blob/master/src/macros/matrix_eq.rs#L333). At the moment it's not a problem though